src/VehicleSetup/FirmwareUpgrade.qml

```diff
@@ -139,12 +139,18 @@ QGCView {
         QGCViewDialog {
             anchors.fill: parent
  
-            property bool showVersionSelection: apmFlightStack.checked || advancedMode.checked
+            property bool showFirmwareTypeSelection: advancedMode.checked
             property bool px4Flow:              controller.boardType == "PX4 Flow"
 
             function accept() {
                 hideDialog()
-                controller.flash(firmwareVersionCombo.model.get(firmwareVersionCombo.currentIndex).firmwareType)
+                var stack = apmFlightStack.checked ? FirmwareUpgradeController.AutoPilotStackAPM : FirmwareUpgradeController.AutoPilotStackPX4
```

Comment:
Stack is always either APM or PX4. Yet flow and radio are marked with their own stack.